pkg/manifests/nginx.go

```diff
@@ -417,9 +417,6 @@ func newNginxIngressControllerDeployment(conf *config.Config, ingressConfig *Ngi
 	}
 
 	podAnnotations := map[string]string{}
-	if !conf.DisableOSM {
-		podAnnotations["openservicemesh.io/sidecar-injection"] = "enabled"
```

Comment:
If this is actually necessary you need to explicitly set it instead of removing it so that this is reconciled for existing resources